facebook/hhvm
# Add a VS Code workspace for building HHVM with Clang

This PR adds the Clang development environment for VS Code, including:
1.  The Nix Flake development shell `devShells.clang` is added to `flake.nix`.
2. The Nix 1.x  development shell file `shell_clang.nix` is added as a wrapper of `devShells.clang`
3. The VS Code workspace file `clang.code-workspace` which uses `shell_clang.nix` to set up environment variables

Test Plan:

In a GitHub Codespace:

- Run `nix develop "git+file://$(pwd)?submodules=1&shallow=1#clang"` to enter the Nix Flake development shell
- Run `nix-shell shell_clang.nix` to the Nix 1.x development shell.
- Open `clang.code-workspace`, and launch a CMake build, then the build should be configured to use clang and libc++
